src/if_checks.c

```diff
@@ -325,3 +325,62 @@ struct check_result *check_name_required_but_not_used_in_if(const struct
 	free_string_list(names_to_check);
 	return res;
 }
+
+struct check_result *check_required_declaration_own(const struct
+                                                    check_data *data,
+                                                    const struct
+                                                    policy_node *node)
+{
+	if (data->flavor != FILE_IF_FILE) {
+		return NULL;
+	}
+
+	const char *name = node->data.d_data->name;
+	const enum decl_flavor flavor = node->data.d_data->flavor;
+
+	// ignore class, permission and user declarations
+	if (flavor == DECL_CLASS || flavor == DECL_PERM || flavor == DECL_USER) {
+		return NULL;
+	}
+
+	// TODO: handle templated declarations
+	if (name[0] == '$') {
+		return NULL;
+	}
+
+	// only check declarations in require blocks
+	for (const struct policy_node *cur = node; cur; cur = cur->parent) {
```

Comment:
Is the interface/template bit essential?  Why not just use is_in_require() from tree.h?

Reply:
switched to `is_in_require()`